### PR TITLE
correctly turn off encryptor

### PR DESCRIPTION
**Pull request by Lagovas**
don't initialize encryptor if config is empty and do not add it to OnQuery observers. it helps to avoid extra parsing of incoming queries

### CODE REVIEW

**Comment by vixentael** [comment]
Can we test it somehow?

**Comment by Lagovas** [comment]
I will add tests soon to this pr

File: decryptor/mysql/proxy_test.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+func TestEncryptorTurnOnOff(t *testing.T) {
+	emptyStore := &tableSchemaStore{true}
+	nonEmptyStore := &tableSchemaStore{false}
+	setting := base.NewProxySetting(&decryptorFactory{}, emptyStore, nil, nil, nil)
+	proxyFactory, err := NewProxyFactory(setting)
+	if err != nil {
+		t.Fatal(setting)
+	}
+	proxy, err := proxyFactory.New(context.TODO(), nil, nil, nil)
```
Comment (storojs72) [inline decryptor/mysql/proxy_test.go]: Why did you create new proxy factory with `context.TODO()` in mysql and with nil in postgres?

Comment (storojs72) [reply decryptor/mysql/proxy_test.go]: It turns out that you need sometimes to pass some context when you create proxy in mysql case, but not in postgres right?

Comment (Lagovas) [reply decryptor/mysql/proxy_test.go]: yes. mysql proxy implementation initialize logger in constructor - https://github.com/cossacklabs/acra/blob/master/decryptor/mysql/response_proxy.go#L175
but postgresql just save context (https://github.com/cossacklabs/acra/blob/master/decryptor/postgresql/pg_decryptor.go#L109) and create logger after (https://github.com/cossacklabs/acra/blob/master/decryptor/postgresql/pg_decryptor.go#L127, https://github.com/cossacklabs/acra/blob/master/decryptor/postgresql/pg_decryptor.go#L481)

**Pull request merged**